update ci builds

CvH opened the pull request:
blind changes, can't test locally

- remove EOL distros besides old TLS
- reorder releases by date
- add new build targets Ubuntu 23.04, 23.10 and Debian Trixie
- left out i386 for new targets, Ubuntu already dropped 32bit

not sure if Debian 13 i386 is needed ?

Flole998 commented:
You're missing the files in the Autobuild directory. Also please squash your commits, this only needs to be a single one.

Flole998 commented:
Also everything that still builds should be left in place. It can still be dropped later on if necessary.

CvH commented:
- added trixie i386 (its already have a autobuild so guess its wanted)
- added autobuild for lunar and mantic

> Also everything that still builds should be left in place.

just kinetic is dropped

> please squash your commits

I squash as soon you are happy with it 👍  (easier to review)

Flole998 commented:
Alright, 2 more things: Disabling those features in the armv7l file should not be done unless it is necessary (we need to check that, on debian bookworm it doesn't seem to be necessary for example). And the same changes need to be done to the build-cloudsmith file aswell so we are providing packages for these new distros.

CvH commented:
- dropped kinetic + added trixi to cloudsmith (blindly, no idea if correct)
- temp removed the disable options from lunatic and mantic

CvH commented:
all green, good to go (i would squash everything before) ?

Flole998 commented:
One thing I just noticed: Please don't delete the kinetic files from the Autobuild directory. If someone wants to build Tvheadend on kinetic it will still be useful for them. As it still builds on a properly setup system those files should be left in place. Other than that, please squash your changes and I will merge them.

CvH commented:
done
